core/commonTest/src/kotlinx/serialization/SerializersLookupNamedCompanionTest.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("RUNTIME_ANNOTATION_NOT_SUPPORTED")
+
+package kotlinx.serialization
+
+import kotlinx.serialization.builtins.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.internal.*
+import kotlin.reflect.*
+import kotlin.test.*
+
+class SerializersLookupNamedCompanionTest {
+    @Serializable
+    class Plain(val i: Int) {
+        @NamedCompanion
```

Comment:
Please also add a test without explicit annotation with assertion that would fail when compiler is actually updated (like it was with `EnumDescriptorTest.assertFullyAnnotated`)